DESIGN.md

```diff
@@ -0,0 +1,63 @@
+
+## Implementation
+
+ts-odd is written in Typescript, a Javascript dialect. The resulting code should be usable in any modern Javascript environment that supports the [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API) and [Web Assembly](https://webassembly.org/). It uses the Rust implementations of [WNFS](https://github.com/wnfs-wg/rs-wnfs) and [UCAN](https://github.com/ucan-wg/rs-ucan) which compile to Web Assembly.
+
+Various other pieces are layered on top of these core protocols and assumptions, these are called "components" in ts-odd. They allow us to adapt to various technologies and environments. We'll start with the components that are closest to the data and go from there.
+
+### Depot
+
+The 'WNFS' file system that holds our user's data consists out of [IPLD](https://ipld.io/) blocks. These blocks make up the content-adressable web, another core idea to the ODD stack, making our data tamper-proof.
+
+This component is reponsible for retrieving and storing these blocks. On the lowest level this means for any given bytes (data) give me back a CID (content address). And vice versa, when I present a CID, give me back the bytes for it.
+
+Looking at this from a higher viewpoint, this unlocks the larger part of data syncing, we can transfer our blocks to a remote source.
+
+### Account
+
+Most of the time to push the data to a remote endpoint you will need some sort of account system, otherwise anyone could fill up the external storage space which costs money. The user accounts will be represented through authorisation, UCANs, our authorisation tickets.
+
+In order to use ts-odd you'll be creating a `Program`, this is a specific setup of components, an artificial environment if you will. This program has a UCAN store, a repository of "authorisation tickets" collected from various sources. The account system is one of those sources. It can also access that UCAN store and then use it to authorise various actions.
+
+This may also tie into the other part of data syncing, taking the file system's root CID (aka. data root) and storing it somewhere the other devices can reach it.
+
+### Identifier & Agent
+
+The account system also involves the identifier and agent components. The identifier signifies one of the user's identifiers and the agent delegates to external services. Both of these use decentralised identifiers ([DID](https://www.w3.org/TR/did-core/)s).
+
+Typically the agent will use a temporary session key pair and the identifier is the more permanent identifier. Example flow: Identifier delegates to agent, agent contacts remote account service, account service issues UCANs addressed to identifier. Those UCANs are then used throughout the SDK to check for capabilities, etc.
```

Comment:
worth spelling out the identifier = passkey or email etc, agent = browser instance hierarchy.

Reply:
Isn't that what the first sentence is saying? I should also say that this is all configurable, so that may not necessarily be the case. But ideally agent = temp key pair, and identifier = permanent or as close as possible